cpp/src/cwrapper/tsfile_cwrapper_expression.cc

```diff
@@ -0,0 +1,207 @@
+                                const char* char_value) {
+    Expression exp;
+    CONSTRUCT_EXP_INTERNAL(exp, column_name);
+    exp.const_condition.value_condition = reinterpret_cast<int64_t>(char_value);
+    exp.const_condition.type = TSDataType::TS_DATATYPE_TEXT;
+    return exp;
+}
+
+TimeFilterExpression* create_andquery_timefilter() {
+    storage::Expression* exp = new storage::Expression(storage::AND_EXPR);
+    return (TimeFilterExpression*)exp;
+}
+
+TimeFilterExpression* create_time_filter(const char* table_name,
+                                         const char* column_name,
+                                         OperatorType oper, int64_t timestamp) {
+    std::string table_name_str(table_name);
+    std::string column_name_str(column_name);
+    storage::Path path(table_name_str, column_name_str);
+    storage::Filter* filter;
+    switch (oper) {
+        case GT:
+            filter = storage::TimeFilter::gt(timestamp);
+            break;
+        case LT:
+            filter = storage::TimeFilter::lt(timestamp);
+            break;
+        case EQ:
+            filter = storage::TimeFilter::eq(timestamp);
+            break;
+        case NOTEQ:
+            filter = storage::TimeFilter::not_eqt(timestamp);
+            break;
+        case GE:
+            filter = storage::TimeFilter::gt_eq(timestamp);
+            break;
+        case LE:
+            filter = storage::TimeFilter::lt_eq(timestamp);
+            break;
+        default:
+            filter = nullptr;
+            break;
+    }
+    storage::Expression* exp =
+        new storage::Expression(storage::SERIES_EXPR, path, filter);
+    return (TimeFilterExpression*)exp;
+}
+
+TimeFilterExpression* add_time_filter_to_and_query(
+    TimeFilterExpression* exp_and, TimeFilterExpression* exp) {
+    storage::Expression* and_exp = (storage::Expression*)exp_and;
+    storage::Expression* time_exp = (storage::Expression*)exp;
+    if (and_exp->left_ == nullptr) {
+        and_exp->left_ = time_exp;
+    } else if (and_exp->right_ == nullptr) {
+        and_exp->right_ = time_exp;
+    } else {
+        storage::Expression* new_exp =
+            new storage::Expression(storage::AND_EXPR);
+        new_exp->left_ = and_exp->right_;
+        and_exp->right_ = new_exp;
+        add_time_filter_to_and_query((TimeFilterExpression*)new_exp, exp);
+    }
+    return exp_and;
+}
+
+void destory_time_filter_query(TimeFilterExpression* expression) {
+    if (expression == nullptr) {
+        return;
+    }
+
+    destory_time_filter_query(
+        (TimeFilterExpression*)((storage::Expression*)expression)->left_);
+    destory_time_filter_query(
+        (TimeFilterExpression*)((storage::Expression*)expression)->right_);
+    storage::Expression* exp = (storage::Expression*)expression;
+    if (exp->type_ == storage::ExpressionType::SERIES_EXPR) {
+        delete exp->filter_;
+    } else {
+        delete exp;
+    }
```

Comment:
'exp' is not deleted in the first branch?
Is it necessary to make this judge? Can you let it delete its 'filter_' in its destruction?